delta-lake/delta-33x/src/main/scala/com/nvidia/spark/rapids/delta/delta33x/GpuDelta33xParquetFileFormat.scala

```diff
@@ -244,4 +273,403 @@ case class GpuDelta33xParquetFileFormat(
+    }
+
+    override def putNotNulls(x$1: Int, x$2: Int): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def putNull(x$1: Int): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def putNulls(x$1: Int, x$2: Int): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def putShort(x$1: Int, x$2: Short): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def putShorts(x$1: Int, x$2: Int, x$3: Array[Byte], x$4: Int): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def putShorts(x$1: Int, x$2: Int, x$3: Array[Short], x$4: Int): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def putShorts(x$1: Int, x$2: Int, x$3: Short): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def reserveInternal(x$1: Int): Unit = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def reserveNewColumn(
+      x$1: Int,
+      x$2: org.apache.spark.sql.types.DataType): WritableColumnVector = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getBoolean(x$1: Int): Boolean = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getByte(x$1: Int): Byte = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getDouble(x$1: Int): Double = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getFloat(x$1: Int): Float = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getInt(x$1: Int): Int = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getLong(x$1: Int): Long = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def getShort(x$1: Int): Short = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def isNullAt(x$1: Int): Boolean = {
+      throw new UnsupportedOperationException()
+    }
+
+    override def close(): Unit = {
+      builder.close()
+    }
+  }
+
+  @scala.annotation.nowarn(
+    "msg=method readFooter in class ParquetFileReader is deprecated"
+  )
+  private def replaceBatch(
+    hadoopConf: Configuration,
+    rowIndex: Long,
+    batch: ColumnarBatch,
+    size: Int,
+    useMetadataRowIndex: Boolean,
+    rowIndexColumnOpt: Option[ColumnMetadata],
+    isRowDeletedColumnOpt: Option[ColumnMetadata],
+    rowIndexFilterOpt: Option[RowIndexFilter],
+    file: PartitionedFile): ColumnarBatch = {
+
+    val (rowIndexCol, isRowDeletedVector) =
+      withResource(new RapidsHostColumnBuilder(new BasicType(false, DType.INT8), size)) { builder =>
+      val isRowDeletedVector = RapidsHostWriteableVector(builder, size, ByteType)
+      val rowIndexCol: RapidsHostColumnVector = if (useMetadataRowIndex) {
+        val rowIndexCol = getRowIndexPos(size, ParquetFileReader.readFooter(hadoopConf,
```

Comment:
This is rereading/reparsing the footer for each batch?

Reply:
This is not needed in the case where we set `useMetadataRowIndex=false`